### DISCUSSION TITLE
Publish to PubSub topic for all events

### DISCUSSION THREAD

**Discussion by brian-liegl-sada** [drakkan/sftpgo#278] [General]
We are looking at a potential integration of SFTPGo into our process, and we will need some assistance in the best way to integrate writing to a pubsub topic in GCS on all events including failed events.

Any thoughts?

**Comment by drakkan** [comment]
Hi,

you can use several hooks to receive the events, take a look here:

https://github.com/drakkan/sftpgo/blob/master/docs/custom-actions.md
https://github.com/drakkan/sftpgo/blob/master/docs/post-connect-hook.md
https://github.com/drakkan/sftpgo/blob/master/docs/post-login-hook.md

it depends on the type of events you are interested in.

Other events could be added, for example I plan to add the `pre-*` event to allow to deny some actions.

You could receive the events you are interested in via HTTP hooks and then forward them to the pubsub topic

**Reply by brian-liegl-sada** [reply]
Does SFTPGo support calling a python script with the post and pre hooks?

**Reply by drakkan** [reply]
> Does SFTPGo support calling a python script with the post and pre hooks?

Yes, it can call any external script that the OS can execute

**Reply by brian-liegl-sada** [reply]
Just a quick update I created a daemon in python that will monitor the log file and then post to pubsub based on events (Upload, Download, Delete, etc.)

**Reply by drakkan** [reply]
yes, this is a way, you could also get HTTP notifications for each event or an external script executed
